packages/api-gateway/src/services/agreementService.ts

```diff
@@ -42,40 +51,42 @@ export function agreementServiceBuilder(
 ) {
   return {
     getAgreements: async (
-      { logger, headers }: WithLogger<ApiGatewayAppContext>,
+      ctx: WithLogger<ApiGatewayAppContext>,
       queryParams: apiGatewayApi.GetAgreementsQueryParams
     ): Promise<apiGatewayApi.Agreements> => {
       const { producerId, consumerId, eserviceId, descriptorId, states } =
         queryParams;
 
-      logger.info(
+      ctx.logger.info(
         `Retrieving agreements for producerId ${producerId} consumerId ${consumerId} eServiceId ${eserviceId} descriptorId ${descriptorId} states ${states}`
       );
 
       if (producerId === undefined && consumerId === undefined) {
         throw producerAndConsumerParamMissing();
       }
 
-      return await getAllAgreements(
-        agreementProcessClient,
-        headers,
-        queryParams
-      );
+      return await getAllAgreements(agreementProcessClient, ctx, queryParams);
     },
 
     getAgreementById: async (
       { logger, headers }: WithLogger<ApiGatewayAppContext>,
       agreementId: agreementApi.Agreement["id"]
     ): Promise<apiGatewayApi.Agreement> => {
       logger.info(`Retrieving agreement by id = ${agreementId}`);
-      const agreement = await agreementProcessClient.getAgreementById({
-        headers,
-        params: {
-          agreementId,
-        },
-      });
+      const agreement = await agreementProcessClient
+        .getAgreementById({
+          headers,
+          params: {
+            agreementId,
+          },
+        })
+        .catch((res) => {
+          throw clientStatusCodeToError(res, {
+            404: agreementNotFound(agreementId),
+          });
```

Comment:
Since it seems it will be always one error code and an error maybe the map is not necessary but I think it's ok even like this.

Reply:
Yeah in this PR it's true, but there are other cases where different errors are caught e.g., in `upsertTenant` I need to throw different errors in case of `404` or `409`